apps/web/src/actions/documents/getContentByPath.ts

```diff
@@ -14,9 +15,13 @@ export const getDocumentContentByPathAction = withProject
     }),
     { type: 'json' },
   )
-  .handler(async ({ input }) => {
+  .handler(async ({ input, ctx }) => {
+    const commitsScope = new CommitsRepository(ctx.project.workspaceId)
+    const commit = await commitsScope
+      .getCommitById(input.commitId)
+      .then((r) => r.unwrap())
     const document = await getDocumentByPath({
-      commitId: input.commitId,
+      commit,
```

Comment:
as a general rule make all db readers/writers receive objects instead of identifiers. This will force the consumer to ensure tenancy and will reduce the amount of redundants roundtrips to the database. It results in fatter controllers but that is ok, tenancy is ensured at the entrypoints of our backend.